### PR TITLE
mpd 0.23.13

### PR DESCRIPTION
**Pull request by chenrui333**
Created by `brew bump`

---

Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by chenrui333** [comment]
```
  ==> meson install -C output/release
  ninja: Entering directory `/tmp/mpd-20230522-55927-9tgyy0/MPD-0.23.13/output/release'
  [1/3] Generating GitVersion.cxx with a custom command
  Traceback (most recent call last):
    File "/home/linuxbrew/.linuxbrew/opt/python@3.11/lib/python3.11/site-packages/mesonbuild/mesonmain.py", line 194, in run
      return options.run_func(options)
             ^^^^^^^^^^^^^^^^^^^^^^^^^
    File "/home/linuxbrew/.linuxbrew/opt/python@3.11/lib/python3.11/site-packages/mesonbuild/minstall.py", line 835, in run
      installer.do_install(datafilename)
    File "/home/linuxbrew/.linuxbrew/opt/python@3.11/lib/python3.11/site-packages/mesonbuild/minstall.py", line 547, in do_install
      self.install_data(d, dm, destdir, fullprefix)
    File "/home/linuxbrew/.linuxbrew/opt/python@3.11/lib/python3.11/site-packages/mesonbuild/minstall.py", line 620, in install_data
      if self.do_copyfile(fullfilename, outfilename, makedirs=(dm, outdir)):
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "/home/linuxbrew/.linuxbrew/opt/python@3.11/lib/python3.11/site-packages/mesonbuild/minstall.py", line 422, in do_copyfile
      self.copy2(from_file, to_file)
    File "/home/linuxbrew/.linuxbrew/opt/python@3.11/lib/python3.11/site-packages/mesonbuild/minstall.py", line 322, in copy2
      shutil.copy2(*args, **kwargs)
    File "/home/linuxbrew/.linuxbrew/opt/python@3.11/lib/python3.11/shutil.py", line 436, in copy2
      copyfile(src, dst, follow_symlinks=follow_symlinks)
    File "/home/linuxbrew/.linuxbrew/opt/python@3.11/lib/python3.11/shutil.py", line 258, in copyfile
      with open(dst, 'wb') as fdst:
           ^^^^^^^^^^^^^^^
  PermissionError: [Errno 13] Permission denied: '/usr/lib/systemd/user/mpd.socket'

```

File: Formula/mpd.rb [hunk 1]
```diff
@@ -2,12 +2,11 @@ class Mpd < Formula
...
   head "https://github.com/MusicPlayerDaemon/MPD.git", branch: "master"
 
   stable do
-    url "https://github.com/MusicPlayerDaemon/MPD/archive/refs/tags/v0.23.12.tar.gz"
-    sha256 "592192b75d33e125eacef43824901cab98a621f5f7f655da66d3072955508c69"
+    url "https://github.com/MusicPlayerDaemon/MPD/archive/refs/tags/v0.23.13.tar.gz"
+    sha256 "c002fd15033d791c8ac3dcc009b728b0e8440ed483ba56e3ff8964587fe9f97d"
     # Add support for fmt 10.0.0 on the v0.23.x branch
     # See https://github.com/MusicPlayerDaemon/MPD/commit/1690c2887f31f45bc5aee66e6283dd4bf338197c
     patch do
```
Comment (carlocab) [inline Formula/mpd.rb:12]: I think we can remove this patch.

File: Formula/mpd.rb
Comment (carlocab) [inline Formula/mpd.rb:1]: I think we need to set `systemd_system_unit_dir` and/or `systemd_user_unit_dir` to fix the build on Linux. Not sure of the correct values, though. CC @Homebrew/linux

Comment (SMillerDev) [reply Formula/mpd.rb:1]: If we depend on systemd we can know where that is. Now it'll just be guessing which you can see here https://unix.stackexchange.com/questions/224992/where-do-i-put-my-systemd-unit-file

Comment (iMichka) [reply Formula/mpd.rb:1]: Agreed to depend on `systemd`, we do this in some occasions and this should be fine. Please rerun this with build with `systemd` added, and see if we need to manually define paths or not, but I think we don't need to.

Comment (carlocab) [reply Formula/mpd.rb:1]: Same error with a `systemd` dependency:
```
  PermissionError: [Errno 13] Permission denied: '/usr/lib/systemd/user/mpd.socket'
```
https://github.com/Homebrew/homebrew-core/actions/runs/5088529656/jobs/9145170561?pr=131617#step:4:882

Can we point these values to the right place now?

**Comment by iMichka** [comment]
I think that we actually don't need to depend on `systemd`. We just need to find a place to write these files to.
There are some explanations here: https://wiki.archlinux.org/title/systemd/User
I decided to put the file in opt_share, and users can copy it to wherever they want to make it work on their system.

**Comment by thezeroalpha** [comment]
In the meantime, you can make the changes yourself (e.g. see [attached diff](https://github.com/Homebrew/homebrew-core/files/11592211/mpd.patch), edit the file with `brew edit mpd` or just apply the patch) and install with `HOMEBREW_NO_INSTALL_FROM_API=1 brew upgrade mpd`.

**Review by carlocab — APPROVED** [review]

File: Formula/mpd.rb [hunk 2]
```diff
@@ -76,10 +70,12 @@ def install
       -Dshout=enabled
       -Dupnp=pupnp
       -Dvorbisenc=enabled
+      -Dsystemd_system_unit_dir=#{lib}/systemd/system
+      -Dsystemd_user_unit_dir=#{lib}/systemd/user
```
Comment (carlocab) [inline Formula/mpd.rb:74]: From reading the `meson.build` scripts, this seems to be where they want to put this stuff.

File: Formula/mpd.rb [hunk 3]
```diff
@@ -54,6 +44,10 @@ class Mpd < Formula
 
   uses_from_macos "curl"
 
+  on_linux do
+    depends_on "systemd" => :build
```
Comment (carlocab) [inline Formula/mpd.rb:48]: `meson.build` seems to look for a `systemd` installation during build time.

**Review by SMillerDev — APPROVED** [review]

**Pull request merged**